### PR TITLE
Add email templates table

### PR DESCRIPTION
**Pull request by amritpurshotam**
Resolves issue #440

### CODE REVIEW

**Comment by amritpurshotam** [comment]
Looks like the test failed due to this error. I think this is fixed by re-running the build. Don't think I have access to do that 
```
Pulling db (orchardup/postgresql:la****)...
ERROR: unexpected EOF
```

File: api/app/email_template/repository.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from app.email_template.models import EmailTemplate
+
+class EmailRepository():
+
+    @staticmethod
+    def get_by_id(email_id):
+        return db.session.query(EmailTemplate).get(email_id)
+
+    @staticmethod
+    def get_by_key(key):
```
Comment (avishkar58) [inline api/app/email_template/repository.py]: Should be by key and event_id

Comment (amritpurshotam) [reply api/app/email_template/repository.py]: Oh yes of course 🙈
Fixed now.

**Comment by amritpurshotam** [comment]
Regarding the null event_id, I think that's a great idea. Gives the best of both worlds then. The event organisers can use the existing templates if they think it's suitable or they can use their own custom ones.
Also just so we're on the same page, is the key essentially an email type? So would example keys look like `PasswordReset`, `EventOffer`, `InvitationLetter`, etc. And in code at the appropriate place, let's say in the password reset flow, I'll search for the `event_id `and `PasswordReset`? And then send the email that's retrieved.

**Comment by avishkar58** [comment]
Yes exactly, although the specific example of `PasswordReset` isn't tied to an event, so we can leave that one (and the email verification ones in the user API) hard-coded. Everything else you'd retrieve by key and event_id when needed :)

File: api/app/email_template/repository.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from app import db
+
+from app.email_template.models import EmailTemplate
+
+class EmailRepository():
+
+    @staticmethod
+    def get(event_id, key):
+        return db.session.query(EmailTemplate).filter_by(event_id=event_id, key=key).first()
```
Comment (avishkar58) [inline api/app/email_template/repository.py]: Think it's best to implement the fall-back logic here

Comment (amritpurshotam) [reply api/app/email_template/repository.py]: Done

**Comment by amritpurshotam** [comment]
Cool. I saw your comment in the other email template issue. Makes sense 👍

**Pull request merged**